tests/lib.rs

```diff
@@ -46,8 +46,8 @@ fn data_map_content_only(){
   assert!(dm.has_chunks() == false);
 }
 
-fn random_string(length: u64) -> String {
-  (0..length).map(|_| (0x20u8 + (rand::random::<f32>() * 96.0) as u8) as char).collect()
+fn random_bytes(length: u64) -> Vec<u8> {
+  (0..length).map(|_| (0x20u8 + (rand::random::<f32>() * 96.0) as u8) as u8).collect()
```

Comment:
Same here, also it looks like most of the code could just take `usize` instead of `u64` as you can't have a `Vec` with > `usize::MAX` elements anyway.